docs/dev/architecture.md

```diff
@@ -93,7 +96,148 @@ The AWS Deadline Cloud Worker Agent has the following key responsibilities:
+*   virtual machines
+*   containers
+
+### 2.1. Worker Agent Installation
+
+The installation of the worker agent is a two step process. First, the Python package must be
+downloaded. An example for this using `pip`:
+
+```sh
+pip install dealdine-cloud-worker-agent
+```
+
+After the Python package is installed, the worker agent installer can be run to complete the setup.
+The Python package provides a `install-deadline-worker` command-line entrypoint for this purpose.
+To learn about the available arguments, run:
+
+```sh
+install-deadline-worker --help
+```
+
+The `install-deadline-worker` program:
+
+- Creates necessary directories and file structures required by the worker agent at run-time
+- Creates required operating system users and groups
+- Sets up required file-system permissions and ownership
+- Grants the required operating system user privileges
+- Sets up operating system services for automatic startup and crash recovery
+
+### 2.2. Operating System Users and Groups
+
+-   **Agent User**:
+    -   Dedicated system user that the worker agent process runs as
+    -   Defaults to `deadline-worker-agent` on Linux and `deadline-worker` on Windows
+    -   Limited privileges following the principle of least privilege
+    -   No login shell access for security
+
+-   **Job Users**:
+    -   Queues can specify a "job run as user" that jobs will run as.
+    -   The worker host admin can configure this user's permissions on the worker host as required.
+    -   Separate user contexts for jobs from different queues
+    -   The job user's primary group is used for managing file-system access to an individual queue job user
+    -   See [Create job users and groups](https://docs.aws.amazon.com/deadline-cloud/latest/developerguide/worker-host.html#create-job-user-and-group) for more information
+
+-   **User Groups**:
+    -   User group that all job users belong to. Used for controlling access to file-system directories accessible by all job users. Defaults to `deadline-job-users`.
+
+### 2.3. File System Structure
+
+The worker agent uses several directories for configuration, state management, and session execution. This section details the purpose and security characteristics of each location.
+
+#### 2.3.1. Configuration Directory
+
+See the [Configuration documentation](../configuration.md).
+
+- **Purpose**: Contains the `worker.toml` configuration file that informs worker agent behavior
+- **Created**: During `install-deadline-worker` execution
+- **Access**:
+  - Read by worker agent user
+  - Written by worker host administrator during installation or for manual configuration
+
+#### 2.3.2. Persistence/Cache Directory
+
+See [State documentation](../state.md).
+
+- **Purpose**: Stores worker identity, credentials, and queue-specific information across worker agent program restarts, and used for sharing queue AWS credentials with session action subprocesses.
+- **Created**: During `install-deadline-worker` execution
```

Comment:
Some of the files are created during worker startup, or during session runs. Should we clarify this?

Reply:
Good catch. I updated this to break out each location under the state directory.